docs/12-self-hosting/03-host-creation/03-QEMU/03-linux-liveiso.mdx

```diff
@@ -0,0 +1,145 @@
+    - https://cdimage.debian.org/debian-cd/current/amd64/iso-dvd/debian-12.1.0-amd64-DVD-1.iso
+  ```
+
+- Download the ISO file
+
+  ```bash
+  export IMAGE_URL="https://cdimage.debian.org/debian-cd/current/amd64/iso-dvd/debian-12.1.0-amd64-DVD-1.iso"
+  export IMAGE_NAME=$(basename -- "$IMAGE_URL")
+  wget -c -O "$IMAGE_NAME" "$IMAGE_URL" -q --show-progress
+  ```
+
+## VM Setup
+
+- Configure the Virtual Machine options
+
+  ```bash
+  # The name of the Virtual Machine
+  export VM_NAME="gameci"
+
+  # Number of physical CPU cores to allocate to the VM
+  export PHYSICAL_CORES="2"
+
+  # Number of threads per core.
+  # Set this to `1` for CPUs that do not support hyper-threading
+  export THREADS="1"
+  export SMP=$(( $PHYSICAL_CORES * $THREADS ))
+
+  # Amount of Disk Space to allocate to the VM.
+  # Cannot exceed available on host.
+  export DISK_SIZE="32G"
+
+  # Amount of RAM to allocate to the VM.
+  # Cannot exceed available RAM on host.
+  export MEMORY="8G"
+
+  # IP address where host may be reached. Do not use `localhost`.
+  export HOST_ADDRESS="SOME IP HERE"
+
+  # Port used by SSH on the host
+  export HOST_SSH_PORT="22"
+
+  # Port to use when forwarding SSH to the VM
+  export VM_SSH_PORT="1234"
+
+  # Port number to expose on the host for VNC
+  export VNC_PORT="0"
+  ```
+
+- Create an empty disk where the OS will be installed.
+
+  ```bash
+  qemu-img create -f qcow2 disk.qcow2 $DISK_SIZE &>/dev/null
+  ```
+
+## Create the VM
+
+- Create new guest:
+
+  ```bash
+  sudo qemu-system-x86_64 \
+    -machine accel=kvm,type=q35 \
+    -cpu host,kvm="off",hv_vendor_id="null" \
+    -smp $SMP,sockets=1,cores="$PHYSICAL_CORES",threads="$THREADS",maxcpus=$SMP \
+    -m "$MEMORY" \
+    -cdrom $IMAGE_NAME \
+    -object iothread,id=io \
+    -device virtio-blk-pci,drive=disk,iothread=io \
+    -drive if=none,id=disk,cache=none,format=qcow2,aio=threads,file=disk.qcow2 \
+    -device intel-hda \
+    -device hda-duplex \
+    -serial stdio -vga virtio -parallel none \
+    -device virtio-net-pci,netdev=network \
+    -netdev user,id=network,hostfwd=tcp::"${VM_SSH_PORT}"-:"${HOST_SSH_PORT}" \
+    -bios /usr/share/ovmf/OVMF.fd \
+    -usbdevice tablet \
+    -vnc $HOST_ADDRESS:$VNC_PORT
+  ```
+
+- Boot existing guest:
+
+  ```bash
+  sudo qemu-system-x86_64 \
+    -machine accel=kvm,type=q35 \
+    -cpu host,kvm="off",hv_vendor_id="null" \
+    -smp $SMP,sockets=1,cores="$PHYSICAL_CORES",threads="$THREADS",maxcpus=$SMP \
+    -m "$MEMORY" \
+    -object iothread,id=io \
+    -device virtio-blk-pci,drive=disk,iothread=io \
+    -drive if=none,id=disk,cache=none,format=qcow2,aio=threads,file=disk.qcow2 \
+    -device intel-hda \
+    -device hda-duplex \
+    -serial stdio -vga virtio -parallel none \
+    -device virtio-net-pci,netdev=network \
+    -netdev user,id=network,hostfwd=tcp::"${VM_SSH_PORT}"-:"${HOST_SSH_PORT}" \
+    -bios /usr/share/ovmf/OVMF.fd \
+    -usbdevice tablet \
+    -vnc $HOST_ADDRESS:$VNC_PORT
+  ```
```

Comment:
Not sure if we should change something here, mostly an observation: These two scripts seem very similar, the `- Create new guest:` example only have `    -cdrom $IMAGE_NAME \` as a difference.

This might be a discussion for the future, but there seem to be quite a few scripts in the self-hosted docs. I started wondering if it would be easier to have a repository with scripts that are ready to use instead of relying on users having to copy everything on their own? This can definitely be done in the future btw.

Reply:
Indeed they are similar, the major change is that the ISO is no longer passed into the VM when booting an existing VM to bypass any issues that may be caused if the machine tried to re-install over-top of your existing data.

I tried to approach these docs in a more "Medium Article / Substack blog" style as to absolved GameCI of having to directly miantain/host a fully-fledged piece of software. This was done based on some previous conversations in discord that trended towards gameci NOT taking ownership of new code as much as possible.

As a result, the docs are now written as a blog-style walkthrough that has pulled the important bits out of some existing software.

Heres where all the QEMU examples come from:
- https://github.com/cloudymax/Scrap-Metal/blob/main/virtual-machines/vm.sh

And heres where the Multipass examples come from:
- https://github.com/cloudymax/Scrap-Metal/blob/main/virtual-machines/multipass/multipass.sh


I'm happy to donate the code to GameCI if that's preferable but that's something we'd need to get buy-in for from the rest of the team.